docs(common): Add AI code assistant docs (#2959)
telerik/blazor-docs: ai/mcp-server.md

---
title: MCP Server
page_title: Telerik Blazor MCP Server

> description: 

Reviewer: Generated by Copilot:
```suggestion
description: Telerik Blazor MCP Server integrates AI-powered tools with proprietary context for Blazor productivity and support
```

* An active [DevCraft or Telerik UI for Blazor license](https://www.telerik.com/purchase/blazor-ui) or a [Telerik UI for Blazor trial](https://www.telerik.com/blazor-ui).

## Installation

Use the documentation of your AI-powered MCP client to add the Telerik MCP server to a specific workspace or globally. You can see installation tips and examples for some popular MCP clients below.

The generic settings of the Telerik Blazor MCP server are:

* Server name: `telerik-blazor-assistant`
* Type: `stdio` (standard input/output transport)
* Command: `npx` (the MCP server works through an npm package)
* Supported arguments: `-y`
* npm package name: `@progress/telerik-blazor-mcp`
* You also need to add your [Telerik licence key](slug:installation-license-key) as an `env` parameter in the `mcp.json` file. There are two options:
    * (recommended) Use a `TELERIK_LICENSE_PATH` argument and point to your Telerik license file location.
    * Use a `TELERIK_LICENSE_KEY` argument and paste your Telerik license key. Make sure to [update the license key](slug:installation-license-key#license-key-updates) when necessary.

### Visual Studio

> Refer to [Use MCP servers in Visual Studio](https://learn.microsoft.com/en-us/visualstudio/ide/mcp-servers).

Reviewer: Consider applying this change to the next two sections.
```suggestion
For detailed instructions, refer to [Use MCP servers in Visual Studio](https://learn.microsoft.com/en-us/visualstudio/ide/mcp-servers).
```

>caption .cursor/mcp.json inside your user folder

````JSON.skip-repl
{
  "mcpServers": {
    "telerik-blazor-assistant": {
      "type": "stdio",
      "command": "npx",
      "args": ["-y", "@progress/telerik-blazor-mcp"],
      "env": {
        "TELERIK_LICENSE_PATH": "C:\\Users\\___\\AppData\\Roaming\\Telerik\\telerik-license.txt"
      }
    }
  }
}
````

## Usage

To use the Telerik MCP Server:

1. Start your prompt with one of the following:
    * `telerik`
    * `/telerik`
    * `@telerik`
    * `telerikblazor`
    * `/telerikblazor`
    * `@telerikblazor`
1. Confirm that the Telerik MCP server is used, because this doesn't happen deterministically. Look for a statement in the output, which is similar to:
    * `Running telerik-blazor-assistant` (in VS Code)
    * `Calling MCP tool telerik-blazor-assistant` (in Cursor)

    If the Telerik MCP server is not used even though it's installed and enabled, then try rephrasing your prompt and use another trigger syntax from the list in step 1.

> 1. Grant the Telerik tool a permission to run for this session, workspace, or always.

Reviewer: ```suggestion
1. Grant the Telerik tool permissions to run for this session, workspace, or always.
```
